From 62879f46302516942ec1b5746466ccfd2b51eb48 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 24 Jan 2024 16:24:25 +0100
Subject: [PATCH 469/841] feat(stm32mp2): adjust DDR4 init delays

Adjust DDR4 init delays to conform with AN5727.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ea32d550f6dc40c07aa49aca4ca4fb4a083e9486 feat(stm32mp2): adjust DDR4 init delays]
Change-Id: Idc097199811ed9ea582c557a4bf23c9719184d20
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/354072
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp2/plat_ddr.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp2/plat_ddr.c b/plat/st/stm32mp2/plat_ddr.c
index b337338fe..87cd12d32 100644
--- a/plat/st/stm32mp2/plat_ddr.c
+++ b/plat/st/stm32mp2/plat_ddr.c
@@ -72,8 +72,7 @@ static int pmic_ddr_power_init(enum ddr_type ddr_type)
 		return status;
 	}
 
-	/* could be set via enable_ramp_delay on vpp_ddr */
-	udelay(2000);
+	udelay(1000);
 
 	status = regulator_enable(vdd_ddr);
 	if (status != 0) {
@@ -89,6 +88,9 @@ static int pmic_ddr_power_init(enum ddr_type ddr_type)
 	if (status != 0) {
 		return status;
 	}
+
+	udelay(1000);
+
 #elif STM32MP_LPDDR4_TYPE
 	int status;
 	struct rdev *vdd1_ddr, *vdd2_ddr;
-- 
2.47.2

